### ISSUE TITLE
[BUG] windows python 3.9 checks memouts and hangs

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#4610]
The CI on python 3.9, windows:

* takes a long time to terminate, if it does
* frequently crashes with memouts or other problems

It is unclear why exactly that combination has problems, but it could be due to a problematic estimator that appears only on 3.9 in windows and uses up a lot of memory and compute, and due to the lower specs of the windows VM.

(recall: estimators are distributed across versions for a given operating system, with a periodicity of 3, so an estimator in 3.7 would reappear on 3.10, but 3.9 does not appear elsewhere since the range is 3.7 to 3.11)

**Comment by fkiraly** [comment]
seems like it's coming from `FCNClassifier`?

**Comment by Jay-sanjay** [comment]
Hey ,
I want to contribute to the issue can you it a bit more precisely and provide
 hints to debug it

**Comment by fkiraly** [comment]
unfortunately, that's as precise as I know - CI hangs on occasion, and sometimes there are vague error messages implicating `FCNClassifier`.

You can run tests locally by doing:
```python
from sktime.utils.estimator_checks import check_estimator
from sktime.classification.deep_learning.fcn import FCNClassifier

check_estimator(FCNClassifier, raise_exceptions=True)
```

**Comment by Jay-sanjay** [comment]
As a beginner towards open source contribution  I wanted some hints that how the issue can be fixed

**Comment by Jay-sanjay** [comment]
also what actually does FCNClassifier means

**Comment by fkiraly** [comment]
> As a beginner towards open source contribution I wanted some hints that how the issue can be fixed

Sure, of course.
Please note that this is not labelled a "good first issue", so it's not a recommended beginner issue, since it might get difficult to diagnose.

Good first issues are here: https://github.com/sktime/sktime/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22
(you can also get here by clicking on the "good first issue" button at the top of the issue tracker)

The suggestion, i.e., the hint to get started was the code pasted above - that is how you run the full test suite for the classifier, called `FCNClassifier` (it's the name of an estimator, a classifier)

**Comment by fkiraly** [comment]
@yarnabrina, I notice that windows 3.9 is now *almost always* failing - any good ideas on this?

**Comment by yarnabrina** [comment]
Note: I locally merged `main` branch in PR #4680 after PR #4682 was merged into `main`, and got a `MemoryError` in `3.7`. And I see PR #4689 has another failure in `3.8`, but I am unable to see the logs.

>  Error: The log was not found. It may have been deleted based on retention settings. 

These may not be related to this issue, but adding the observation just in case it is.

**Comment by achieveordie** [comment]
Since this issue seems prolific, we'd probably be better of with excluding all the DL estimators (and perhaps even adding a warning message). I'll try to profile these estimators over the next few days and try to diagnose concrete root causes for it. Right now it seems like every once in a while one of the estimators decides to be grumpy and causes entire machine to crash.

**Comment by fkiraly** [comment]
Yes, that's what I was thinking - this is really getting annoying, very hard to track down. I am also suspecting there might be some weird side effects that keep estimators in memory where they should have been destructed in testing.

Would you like to make that PR to keep the handling of this in one place? I'll be looking at some of the bugfixing backlog meanwhile.

**Comment by fnhirwa** [comment]
The tests kept failing due to `tensorflow` import order error it conflicts with `pandas` and a workaround is to import tensorflow before any other module.

When running the `check_estimator` I had to import `tf` before importing other modules.

```python
import tensorflow as tf

from sktime.classification.deep_learning.mcnn import SomeDLEstimator
from sktime.utils.estimator_checks import check_estimator

check_estimator(SomeDLEstimator)

```

This snippet will work without hanging.

The other handy solution is downgrading the `tensorFlow` to `2.15.0` makes the tests run and examples without hanging.
As suggested from this post [Importing pandas before tensorflow makes the script freeze](https://stackoverflow.com/questions/78561116/importing-pandas-before-tensorflow-makes-the-script-freeze)
I am trying to work on fixes from this PR: https://github.com/sktime/sktime/pull/6763 once it works we can incorporate the fix to all skipped tests.
